src/Servers/Kestrel/Core/src/KestrelConfigurationLoader.cs

```diff
@@ -289,9 +289,11 @@ public void Load()
                         httpsOptions.SslProtocols = endpoint.SslProtocols.Value;
                     }
 
+                    httpsOptions.ClientCertificateMode = endpoint.ClientCertificateMode ?? ConfigurationReader.ClientCertificateMode ?? httpsOptions.ClientCertificateMode;
```

Comment:
I think we should apply the global ConfigurationReader.ClientCertificateMode before calling ApplyHttpsDefaults then apply endpoint.ClientCertificateMode afterwards. This way global config doesn't override what was done in code via ConfigureHttpsDefaults. Our handling of SslProtocols is very similar.

We should also add some KestrelConfigurationLoader tests to verify whatever behavior we decide on.